ethereum/core/src/main/java/org/hyperledger/besu/ethereum/core/Transaction.java

```diff
@@ -80,6 +86,57 @@ public static Builder builder() {
   }
 
   public static Transaction readFrom(final RLPInput input) throws RLPException {
+    if (!ExperimentalEIPs.eip1559Enabled) {
```

Comment:
Style: I think it's premature to call the current code legacy until EIP1559 is adopted.  I think there are two alternatives
* Keep the guard block, but have it switch to `readFromExperimental` and keep the current operations as the main text
* Hav two new methods `readFromFrontier` and `readFromEIP1559` (to be renamed `readFrom<fork>` when assigned) and switch in the current `readFrom` method.

Reply:
Ok. Will go with the first alternative

Reply:
Done